Sources/Playback/Control/VLCPlaybackService.m

```diff
@@ -454,8 +454,12 @@ - (void)restoreAudioAndSubtitleTrack
     VLCMLMedia *media = [_delegate mediaForPlayingMedia:_mediaPlayer.media];
 
     if (media) {
-        _mediaPlayer.currentAudioTrackIndex = (int) media.audioTrackIndex;
-        _mediaPlayer.currentVideoSubTitleIndex = (int) media.subtitleTrackIndex;
+        NSTimeInterval delayInSeconds = 0.2;
+        dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));
+        dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
+            _mediaPlayer.currentAudioTrackIndex = (int) media.audioTrackIndex;
+            _mediaPlayer.currentVideoSubTitleIndex = 3; // (int) media.subtitleTrackIndex;
+        });
```

Comment:
Why the dispatch?

and that constant of 3 is wrong. It can be any number.

Reply:
I believe at the time of restoreAudioAndSubtitleTrack was called, the mediaPlayer was not fully initialized. So added a short delay to ensure the mediaPlayer is fully initialized to perform other setup.

Reply:
#461 was a mistake, just fixed it.

Reply:
@fkuehne please take a look.